Longer timeout limit needed for LeMUR (java.net.SocketTimeoutException: timeout)

Transcripts with longer token size time out with LeMUR functions. 

Requests with longer files result in this error: `Caused by: java.net.SocketTimeoutException: timeout`

For context, here is the code used:

```java
import com.assemblyai.api.AssemblyAI;
import com.assemblyai.api.resources.lemur.requests.*;
import com.assemblyai.api.resources.transcripts.types.*;
import java.util.List;

public final class App {
    public static void main(String[] args) {

        AssemblyAI client = AssemblyAI.builder()
                .apiKey("api-key")
                .build();

        String url = "https://storage.googleapis.com/aai-web-samples/meeting.mp3";

        Transcript transcript = client.transcripts().transcribe(url);

        String prompt = "Provide a brief summary of the transcript.";

        var params = LemurTaskParams.builder()
                .prompt(prompt)
                .transcriptIds(List.of(transcript.getId()))
                .build();

        var result = client.lemur().task(params);

        System.out.println(result.getResponse());
```

--- a/.fernignore
+++ b/.fernignore
@@ -1,5 +1,4 @@
 # Specify files that shouldn't be modified by Fern
-
 src/main/java/com/assemblyai/api/AssemblyAI.java
 src/main/java/com/assemblyai/api/PollingTranscriptsClient.java
 src/main/java/com/assemblyai/api/Transcriber.java
@@ -14,6 +13,7 @@ src/main/java/com/assemblyai/api/resources/realtime/types/SessionBegins.java
 src/main/java/com/assemblyai/api/resources/realtime/types/SessionTerminated.java
 
 # Ignore ClientOptions until support for timeout configuration is added.
+src/main/java/com/assemblyai/api/AssemblyAIBuilder.java
 src/main/java/com/assemblyai/api/core/ClientOptions.java
 
 

--- a/src/main/java/com/assemblyai/api/AssemblyAI.java
+++ b/src/main/java/com/assemblyai/api/AssemblyAI.java
@@ -8,6 +8,7 @@ import com.assemblyai.api.core.Suppliers;
 import com.assemblyai.api.resources.files.FilesClient;
 import com.assemblyai.api.resources.lemur.LemurClient;
 import com.assemblyai.api.resources.realtime.RealtimeClient;
+
 import java.util.function.Supplier;
 
 public class AssemblyAI {
@@ -27,6 +28,14 @@ public class AssemblyAI {
         this.transcriptClient = Suppliers.memoize(() -> new PollingTranscriptsClient(clientOptions, this));
         this.realtimeClient = Suppliers.memoize(() -> new RealtimeClient(clientOptions));
         this.lemurClient = Suppliers.memoize(() -> new LemurClient(clientOptions));
+    }
+
+    public AssemblyAI(ClientOptions clientOptions, ClientOptions lemurClientOptions) {
+        this.clientOptions = clientOptions;
+        this.filesClient = Suppliers.memoize(() -> new FilesClient(clientOptions));
+        this.transcriptClient = Suppliers.memoize(() -> new PollingTranscriptsClient(clientOptions, this));
+        this.realtimeClient = Suppliers.memoize(() -> new RealtimeClient(clientOptions));
+        this.lemurClient = Suppliers.memoize(() -> new LemurClient(lemurClientOptions));
     }
 
     public FilesClient files() {

--- a/src/main/java/com/assemblyai/api/AssemblyAIBuilder.java
+++ b/src/main/java/com/assemblyai/api/AssemblyAIBuilder.java
@@ -7,12 +7,14 @@ import com.assemblyai.api.core.ClientOptions;
 import com.assemblyai.api.core.Environment;
 
 public final class AssemblyAIBuilder {
-    private ClientOptions.Builder clientOptionsBuilder = ClientOptions.builder();
+    private final ClientOptions.Builder clientOptionsBuilder = ClientOptions.builder();
+    private final ClientOptions.Builder lemurClientOptionsBuilder = ClientOptions.builder();
 
     private Environment environment = Environment.DEFAULT;
 
     public AssemblyAIBuilder apiKey(String apiKey) {
         this.clientOptionsBuilder.addHeader("Authorization", apiKey);
+        this.lemurClientOptionsBuilder.addHeader("Authorization", apiKey);
         return this;
     }
 
@@ -28,6 +30,9 @@ public final class AssemblyAIBuilder {
 
     public AssemblyAI build() {
         clientOptionsBuilder.environment(this.environment);
-        return new AssemblyAI(clientOptionsBuilder.build());
+        lemurClientOptionsBuilder
+                .environment(this.environment)
+                .disableTimeouts();
+        return new AssemblyAI(clientOptionsBuilder.build(), lemurClientOptionsBuilder.build());
     }
 }

--- a/src/main/java/com/assemblyai/api/core/ClientOptions.java
+++ b/src/main/java/com/assemblyai/api/core/ClientOptions.java
@@ -7,6 +7,7 @@ import java.util.HashMap;
 import java.util.Map;
 import java.util.concurrent.TimeUnit;
 import java.util.function.Supplier;
+
 import okhttp3.OkHttpClient;
 
 public final class ClientOptions {
@@ -26,16 +27,13 @@ public final class ClientOptions {
         this.environment = environment;
         this.headers = new HashMap<>();
         this.headers.putAll(headers);
-        this.headers.putAll(Map.of(
-                "X-Fern-SDK-Name",
-                "com.assemblyai.fern:api-sdk",
-                "X-Fern-SDK-Version",
-                "1.0.2",
-                "X-Fern-Language",
-                "JAVA"));
+        this.headers.putAll(new HashMap<String, String>() {{
+            put("X-Fern-SDK-Name", "com.assemblyai.fern:api-sdk");
+            put("X-Fern-SDK-Version", "1.0.2");
+            put("X-Fern-Language", "JAVA");
+        }});
         this.headerSuppliers = headerSuppliers;
         this.httpClient = httpClient;
-        ;
     }
 
     public Environment environment() {
@@ -67,6 +65,7 @@ public final class ClientOptions {
         private final Map<String, String> headers = new HashMap<>();
 
         private final Map<String, Supplier<String>> headerSuppliers = new HashMap<>();
+        private boolean disableTimeouts = false;
 
         public Builder environment(Environment environment) {
             this.environment = environment;
@@ -83,13 +82,32 @@ public final class ClientOptions {
             return this;
         }
 
+        /**
+         * This is a temporary measure ot disable timeouts for LeMUR client.
+         * Don't use this method.
+         *
+         * @return ClientOptionsBuilder
+         * @deprecated
+         */
+        public Builder disableTimeouts() {
+            this.disableTimeouts = true;
+            return this;
+        }
+
         public ClientOptions build() {
-            OkHttpClient okhttpClient = new OkHttpClient.Builder()
-                    .addInterceptor(new RetryInterceptor(3))
-                    // set longer timeout for LeMUR
-                    .connectTimeout(30, TimeUnit.SECONDS)
-                    .build();
-            return new ClientOptions(environment, headers, headerSuppliers, okhttpClient);
+            OkHttpClient.Builder okhttpClientBuilder = new OkHttpClient.Builder()
+                    .addInterceptor(new RetryInterceptor(3));
+            if (this.disableTimeouts) {
+                okhttpClientBuilder
+                        .readTimeout(0, TimeUnit.SECONDS)
+                        .callTimeout(0, TimeUnit.SECONDS);
+            }
+            return new ClientOptions(
+                    environment,
+                    headers,
+                    headerSuppliers,
+                    okhttpClientBuilder.build()
+            );
         }
     }
 }

